core/src/main/scala/cats/collections/Heap.scala

```diff
@@ -130,14 +130,20 @@ sealed abstract class Heap[A] {
     Heap.heapify(a)
 
   /**
-   * Remove the min element from the heap (the root).
+   * Remove the min element from the heap (the root) and return it along with the updated heap.
    * Order O(log n)
    */
-  def remove(implicit order: Order[A]): Heap[A] = this match {
-    case Branch(_, l, r) => bubbleRootDown(mergeChildren(l, r))
-    case Leaf()          => Leaf()
+  def pop(implicit order: Order[A]): (Heap[A], Option[A]) = this match {
```

Comment:
I think `Option[(A, Heap[A])]` would be a better signature here, and I think only fewer allocations (since the empty case doesn’t allocate at all) and the non empty case allocated one tuple and one Some just as your implementation does.

Reply:
ok 👍